### PR TITLE
Fix/hardcoded network names

### PR DESCRIPTION
**Pull request by Aierie**
CS-1143

Replaces hardcoded network names with the correct ones identified at build time. For `*.ts` files, this is done by importing an object with references to layer 1 and layer 2 names. For `*.hbs` files, this is done using a helper.

Will wait for pending PRs for prepaid card issuance flow to be merged, then update this before merging it.

### CODE REVIEW

File: packages/web-client/app/components/card-pay/deposit-workflow/index.ts [hunk 1]
```diff
@@ -10,12 +10,14 @@ import Layer1Network from '@cardstack/web-client/services/layer1-network';
...
 import { action } from '@ember/object';
+import { currentNetworkDisplayInfo as c } from '@cardstack/web-client/utils/web3-strategies/network-display-info';
+import { capitalize } from '@ember/string';
 
 class DepositWorkflow extends Workflow {
   name = 'Reserve Pool Deposit';
   milestones = [
     new Milestone({
-      title: 'Connect mainnet wallet',
+      title: `Connect ${c.layer1.networkType} wallet`,
```
Comment (Aierie) [inline packages/web-client/app/components/card-pay/deposit-workflow/index.ts:20]: Unsure about replacing mainnet with testnet.

File: packages/web-client/app/components/card-pay/deposit-workflow/index.ts [hunk 2]
```diff
@@ -52,22 +54,22 @@ class DepositWorkflow extends Workflow {
           componentName: 'card-pay/layer-one-connect-card',
         }),
       ],
-      completedDetail: 'Mainnet wallet connected',
+      completedDetail: `${capitalize(c.layer1.networkType)} wallet connected`,
```
Comment (Aierie) [inline packages/web-client/app/components/card-pay/deposit-workflow/index.ts:57]: Wondering whether this should be the responsibility of the `NetworkCopywriting` type - capitalized vs non-capitalized stuff?

File: packages/web-client/app/utils/web3-strategies/ethereum.ts [hunk 3]
```diff
@@ -1,7 +1,8 @@
 import Layer1ChainWeb3Strategy from './layer1-chain';
+import { networkDisplayInfo } from './network-display-info';
 
 export default class EthereumWeb3Strategy extends Layer1ChainWeb3Strategy {
   constructor() {
-    super('mainnet', 'Ethereum mainnet');
+    super('mainnet', networkDisplayInfo.mainnet.fullName);
```
Comment (backspace) [inline packages/web-client/app/utils/web3-strategies/ethereum.ts:6]: Now that there’s a `networkDisplayInfo` to collect these programmatic and UI names, maybe this second argument to the constructor is unnecessary and can be derived in the constructors for the strategies? 🤔

File: packages/web-client/app/utils/web3-strategies/types.ts [hunk 4]
```diff
@@ -54,3 +54,4 @@ export interface Layer2Web3Strategy extends Web3Strategy {
 
 export type TransactionHash = string;
 export type ChainAddress = string;
+export type NetworkSymbol = 'kovan' | 'sokol' | 'mainnet' | 'xdai';
```
Comment (backspace) [inline packages/web-client/app/utils/web3-strategies/types.ts:57]: I’m happy to see this graduate into this file haha

File: packages/web-client/app/components/card-pay/deposit-workflow/index.ts (hunk 2 above)
Comment (backspace) [reply packages/web-client/app/components/card-pay/deposit-workflow/index.ts:57]: Maybe… semi-relatedly, I anticipate some difficulty around things like “an Ethereum wallet” vs “a Kovan wallet” but I feel like it’s not worth doing anything in that direction until it‘s truly needed, it doesn’t seem like this implementation will hamper updates toward that end if a need does emerge.

File: packages/web-client/app/components/card-pay/deposit-workflow/confirmation/index.hbs [hunk 5]
```diff
@@ -25,7 +25,7 @@
         <CardPay::AccountDisplay
           class="card-pay-confirmation__transaction-wallet"
           @size="small"
-          @name={{concat this.layer2Network.chainName " wallet"}}
+          @name={{concat this.layer1Network.chainName " wallet"}}
```
Comment (backspace) [inline packages/web-client/app/components/card-pay/deposit-workflow/confirmation/index.hbs:28]: 😯

**Review by backspace — APPROVED** [review]
This looks suitable to me, I made an optional suggestion about deriving one of the chain strategy constructor arguments but definitely not blocking. I’m looking forward to seeing the proper strings in the different environments, I think it’ll help reduce my confusion as I still learn 🤓

File: packages/web-client/app/utils/web3-strategies/ethereum.ts (hunk 3 above)
Comment (Aierie) [reply packages/web-client/app/utils/web3-strategies/ethereum.ts:6]: This seems good, it reduces the number of files that need to know about `networkDisplayInfo`.

Comment (Aierie) [reply packages/web-client/app/utils/web3-strategies/ethereum.ts:6]: Will change

File: packages/web-client/app/components/card-pay/deposit-workflow/confirmation/index.hbs (hunk 5 above)
Comment (Aierie) [reply packages/web-client/app/components/card-pay/deposit-workflow/confirmation/index.hbs:28]: Yep, this is the top half for:
image

File: packages/web-client/app/components/card-pay/deposit-workflow/index.ts (hunk 1 above)
Comment (lukemelia) [reply packages/web-client/app/components/card-pay/deposit-workflow/index.ts:20]: Yeah, it is not clear, because both sokol and kovan are testnets.

Comment (Aierie) [reply packages/web-client/app/components/card-pay/deposit-workflow/index.ts:20]: Will check with design about whether we can replace 'mainnet' with the full testnet name.

**Comment by backspace** [comment]
#1704 has “on xDai chain” hardcoded in for now in the prepaid card preview step, if/when it’s merged, could update that here? 🤞🏻

**Comment by Aierie** [comment]
> #1704 has “on xDai chain” hardcoded in for now in the prepaid card preview step, if/when it’s merged, could update that here? 🤞🏻

Yep! Planning to merge this only after your PR.

File: packages/web-client/app/components/card-pay/deposit-workflow/index.ts (hunk 1 above)
Comment (Aierie) [reply packages/web-client/app/components/card-pay/deposit-workflow/index.ts]: Kovan's name will read `Kovan testnet`.

**Pull request merged**